gulpfile.js

```diff
@@ -5,30 +5,121 @@
    its own file in gulp/tasks. Any files in that directory get
    automatically required below.
    To add a new task, simply add a new task file the gulp/tasks directory.
-   gulp/tasks/default.js specifies the default set of tasks to run
-   when you run `gulp`. */
+   To ignore tasks in production environments, add to the ignoreTasks array. */
 
+const envvars = require( './config/environment' ).envvars;
+const fancyLog = require( 'fancy-log' );
+const fs = require( 'fs' );
+const glob = require( 'glob' );
 const gulp = require( 'gulp' );
-const requireDir = require( 'require-dir' );
+const path = require( 'path' );
 
-// Require all tasks in gulp/tasks, including subfolders.
-requireDir( './gulp/tasks', { recurse: true } );
+// Define tasks that should be ignored in production environment.
+const TASK_PATH = './gulp/tasks/';
+let ignoreTasks = [];
 
-gulp.task( 'build',
-  gulp.series(
+if ( envvars.NODE_ENV === 'production' ) {
+  ignoreTasks = [
+    TASK_PATH + 'audit.js',
+    TASK_PATH + 'docs.js',
+    TASK_PATH + 'lint.js',
+    TASK_PATH + 'test-unit.js'
+  ];
+}
+
+/**
+ * Require a gulp task and log the task to the console.
+ * @param  {string} taskPath - Path to a gulp task.
+ */
+function requireTask( taskPath ) {
+  fancyLog( 'Requiring task', taskPath );
+  // eslint-disable-next-line global-require
+  require( path.resolve( taskPath ) );
+}
+
+/**
+ * Recursively check that a file exists on disk, given a file pattern in the
+ * format file-name.
+ * E.g. Take the file pattern test-unit-scripts.
+ * This function will check (recursively)
+ * whether test-unit-scripts, test-unit, and test exist.
+ * If one of those exist, it will return the first occurence,
+ * otherwise it will return undefined.
+ * @param  {string} filePattern A filename pattern in the format file-name.
+ * @returns {string|undefined} File name that exists, otherwise undefined.
+ */
+function fileExists( filePattern ) {
+  let UNDEFINED;
+  if ( !filePattern ) {
+    return UNDEFINED;
+  }
+  const checkFile = `${ TASK_PATH }${ filePattern }.js`;
+  // eslint-disable-next-line no-sync
+  const doesFileExist = fs.existsSync( checkFile );
```

Comment:
Since this isn't reused at all, is it worth moving the execution to the condition? It would clean up the repetitive naming a bit.

Reply:
Fixed